src/activity_handlers.cpp

```diff
@@ -255,9 +255,9 @@ void set_up_butchery( player_activity &act, player &u, butcher_type action )
         return;
     }
 
-    int factor = u.max_quality( quality_id( "BUTCHER" ) );
+    int factor = u.max_quality( action == DISSECT ? quality_id( "CUT_FINE" ) : quality_id( "BUTCHER" ) );
     auto items = g->m.i_at( u.pos() );
-    if( ( size_t )act.index >= items.size() || factor == INT_MIN ) {
```

Comment:
This would print a "No Corpses" error when it returns to butcher_finish. This has been handled by code following this.